### PR TITLE
feat: Initial Auth Service

### LINKED ISSUE #34: Initial Auth Service
Contains server start command using Cobra and controller implementation with an empty handler.

### PR DESCRIPTION
**Pull request by sandrask**
Contains server start command using Cobra and controller implementation with an empty handler.

Closes #34

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/restapi/auth/operation/operations_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// prepare router
+	router := mux.NewRouter()
+
+	router.HandleFunc(handler.Path(), handler.Handle()).Methods(handler.Method())
+
+	// create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
```
[the remark this answers is not in the document]
Comment (DRK3) [reply pkg/restapi/auth/operation/operations_test.go]: (so perhaps these explanatory comments made sense in the tutorial, but aren't needed here?)

File: pkg/restapi/auth/operation/operations.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"io"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/trustbloc/edge-store/pkg/internal/common/support"
+)
+
+const (
+	token = "/token"
```
[the remark this answers is not in the document]
Comment (sandrask) [reply pkg/restapi/auth/operation/operations.go]: I prefer token to token endpoint. Maybe get-token makes more sense. I will re-evaluate name as I develop OAuth flow. Also,  I consolidated common code between servers in utility package and addressed all issues with comments.

File: scripts/check_unit.sh [hunk 3]
```diff
@@ -31,4 +31,17 @@ PKGS=`go list github.com/trustbloc/edge-store/cmd/edge-store-rest/... 2> /dev/nu
...
+
+# Running edge-auth-rest unit tests
+cd cmd/edge-auth-rest
+PKGS=`go list github.com/trustbloc/edge-store/cmd/edge-auth-rest/... 2> /dev/null | \
+                                                 grep -v /mocks`
+go test $PKGS -count=1 -race -coverprofile=profile.out -covermode=atomic -timeout=10m
+amend_coverage_file
 cd "$pwd" || exit
+
+
```
[the remark this answers is not in the document]
Comment (DRK3) [reply scripts/check_unit.sh]: @sandrask I still see the extra whitespace here

[the remark this answers is not in the document]
Comment (sandrask) [reply scripts/check_unit.sh]: fixed

File: pkg/restapi/auth/operation/operations.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (llorllale) [reply pkg/restapi/auth/operation/operations.go]: Oh, I see. I guess I agree but I'm not strongly arguing for/against in the grand scheme of things...

**Pull request merged**